spec/features/rooms/index_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+
+RSpec.describe "index spec" do
+  before(:each) do
+    @emma = Hotel.create!(name: "Hotel Emma", location: "San Antonio")
+    @wolf = Hotel.create!(name: "Great Wolf Lodge", location: "Dallas")
+    
+    @room_1 = @emma.rooms.create!(rate: 500, suite: "Presidential")
+    @room_2 = @emma.rooms.create!(rate: 650, suite: "Penthouse")
+    @room_3 = @wolf.rooms.create!(rate: 300, suite: "Garden")
+    @room_4 = @wolf.rooms.create!(rate: 400, suite: "Animal Kingdom")
+    
+    @sara = Guest.create!(name: "Sara", nights: 2)
+    @leo = Guest.create!(name: "Leo", nights: 3)
+    @piper = Guest.create!(name: "Piper", nights: 3)
+    @paige = Guest.create!(name: "Paige", nights: 1)
+
+    @room_4.guests << @sara
+    @room_1.guests << @sara
+    @room_2.guests << @paige
+    @room_1.guests << @paige
+  end
+#user story 3
+  describe " When I visit the rooms index page" do
+    it "has a list of all rooms and its attributes and its hotel" do
+      visit "/rooms"
+
+      expect(page).to have_content(@room_1.suite)
+      expect(page).to have_content(@room_2.suite)
+      expect(page).to have_content(@room_3.suite)
+      expect(page).to have_content(@room_4.suite)
+      expect(page).to have_content(@room_1.rate)
+      expect(page).to have_content(@room_2.rate)
+      expect(page).to have_content(@room_3.rate)
+      expect(page).to have_content(@room_4.rate)
+      expect(page).to have_content(@room_1.hotel.name)
+      expect(page).to have_content(@room_2.hotel.name)
+      expect(page).to have_content(@room_3.hotel.name)
+      expect(page).to have_content(@room_4.hotel.name)
+    end
+
+    it " show the number of guests that stayed in each room" do 
+      visit "/rooms"
+
+      expect(page).to have_content(@room_1.suite)
+      expect(page).to have_content(@room_1.rate)
+      expect(page).to have_content(@room_1.hotel.name)
+      # Idk how you guys want that tested here so heres 3 ways
+      expect(page).to have_content(@room_1.guests.count)
```

Comment:
The assertion on line 50 is the best because it makes sure that the guest count shows up next to the `Guest Count: ` text, rather than there being a floating 2 randomly on the page. 

It's also best to test in this way rather than verifying that it equals `@room_1.guests.count` because the `guests.count` is what you have in your implementation in the model method. When testing, we don't want to use the same code that's used in the implementation because the calculations in both places will always be the same, but not necessarily indicate whether the functionality is correct or expected. For example, given this method
`def add(a, b)`
`   a + b`
`end`

When we're testing this, we would want to test it like this: `expect(add(1,2)).to eq(3)` instead of `expect(add(1,2)).to eq(1 + 2)`. In the second case, we're just using the same calculation that's in the code. If it's *wrong* and not the right implementation, our test wouldn't catch it because it's just the same calculation in the test. 

Let me know if you have follow up questions!